File: 24. Swap Nodes in Pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        if not head or not head.next:
            return head
        dummyhead = ListNode(0)
        dummyhead.next = head
        prev = dummyhead       
        while head and head.next:
            prev.next = head.next
            temp = head.next.next
            prev = head
            head.next.next = head
            head.next = temp
            head = head.next
        return dummyhead.next

File: 24. Swap Nodes in Pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode next = head.next;
        ListNode node = head;
        head = next;
        ListNode prev = null;
        while (next != null && node != null) {
            ListNode nextnext = next.next;
            if (prev != null) {
                prev.next = next;
            }
            node.next = nextnext;
            next.next = node;
            prev = node;
            if (node.next == null || node.next.next == null) {
                break;
            }
            node = node.next;
            next = node.next;
        }
        return head;
    }
}